### DISCUSSION TITLE
83. 파티셔닝과 샤딩 개념에 대해 설명해주세요.

### DISCUSSION THREAD

**Discussion by cutehumanS2** [jewelry-box/cs-interview#85] [General]
- 파티셔닝과 샤딩의 목적은 무엇이며, 각각의 기술을 어떤 경우에 사용하는 것이 적합한가요?

**Comment by linirini** [comment]
1. **파티셔닝**
- 하나의 데이터베이스 내에서 테이블을 **논리적으로** 나누는 방법
- 목적
  - 대규모 데이터를 처리하거나 관리하기에 용이
  - 성능을 향상시키고 쿼리 속도를 최적화
  - 데이터 분산을 통한 I/O 병목 감소
- 사용 예시
  - 단일 데이터베이스 환경에서 성능 최적화 및 데이터 관리 효율성 증가
  - 주기적으로 특정 조건의 데이터만 쿼리하는 경우
  - I/) 병목 줄이거나 성능 최적화해야하는 경우
2. **샤딩**
- 데이터베이스를 여러 개의 독립적인 **물리적 노드로 분산**하는 방법
- 각 노드는 하나의 샤드로 동작하며, 각각 데이터의 서브셋ㅇ르 저장
- 목적
  - 대규모 데이터 처리를 위한 수평적 확장 지원
  - 단일 데이터베이스의 성능 및 저장 용량 한계 극복
  - 부하를 여러 서버로 분산하여 처리 속도 높이고 가용성 확보
- 사용 예시
  - 단일 데이터베이스 용량 한계 초과 시
  - 수평 확장 필요 시
  - 독립적을 관리 간으한 데이터 세트가 있는 경우
  - 대규모 분산 시스템에서
